golem-worker-service/src/api/openapi/converter.rs

```diff
@@ -0,0 +1,1019 @@
+                parameters: {
+                    let mut params = Self::extract_path_parameters(&route.path).unwrap_or_default();
+                    params.extend(Self::extract_query_parameters(route));
+                    params.extend(Self::extract_header_parameters(route));
+                    if params.is_empty() { None } else { Some(params) }
+                },
+                request_body: Self::create_request_body(route),
+                responses: {
+                    let mut map = Self::create_responses(route, "*");
+                    // Add CORS headers to all responses
+                    for response in map.values_mut() {
+                        response.headers = Some(Self::create_cors_headers("*"));
+                    }
+                    map
+                },
+                security: None,
+                tags: Some(vec![route.template_name.clone()]),
+            },
+            BindingType::Proxy => Operation {
+                summary: Some(route.description.clone()),
+                description: None,
+                operation_id: Some(format!("{}_{}",
+                    route.template_name.to_snake_case(),
+                    route.method.to_string().to_lowercase())),
+                parameters: {
+                    let mut params = Self::extract_path_parameters(&route.path).unwrap_or_default();
+                    params.extend(Self::extract_query_parameters(route));
+                    params.extend(Self::extract_header_parameters(route));
+                    if params.is_empty() { None } else { Some(params) }
+                },
+                request_body: Self::create_request_body(route),
+                responses: {
+                    let mut map = Self::create_responses(route, "*");
+                    // Add CORS headers to all responses
+                    for response in map.values_mut() {
+                        response.headers = Some(Self::create_cors_headers("*"));
+                    }
+                    map
+                },
+                security: None,
+                tags: Some(vec![route.template_name.clone()]),
+            },
+        }
+    }
+
+    fn extract_path_parameters(path: &str) -> Option<Vec<Parameter>> {
+        // Use the official parser to parse the path pattern
+        let path_pattern = match AllPathPatterns::parse(path) {
+            Ok(pattern) => pattern,
+            Err(_) => return None
+        };
+
+        // Extract parameters from path patterns
+        let params: Vec<Parameter> = path_pattern.path_patterns
+            .iter()
+            .filter_map(|pattern| match pattern {
+                PathPattern::Var(info) => Some(Parameter {
+                    name: info.key_name.clone(),
+                    r#in: ParameterLocation::Path,
+                    description: None, 
+                    required: Some(true),
+                    schema: if info.key_name.ends_with("_id") {
```

Comment:
For path parameters, Rib has inferred types, which we should be using to generate the schema.

In general, we will need, in a separate module (so it's easy to test), a way to convert a Rib type into an OpenAPI schema. Then we will use that for the path parameters, query variables, and request / response headers & bodies.